sky/backends/backend_utils.py

```diff
@@ -1506,11 +1506,14 @@ def check_network_connection():
     http = requests.Session()
     http.mount('https://', adapter)
     http.mount('http://', adapter)
-    try:
-        http.head(_TEST_IP, timeout=3)
-    except (requests.Timeout, requests.exceptions.ConnectionError) as e:
-        raise exceptions.NetworkError(
-            'Could not refresh the cluster. Network seems down.') from e
+    for ip in _TEST_IP_LIST:
+        try:
+            http.head(ip, timeout=3)
+            return
+        except (requests.Timeout, requests.exceptions.ConnectionError) as e:
+            if ip == _TEST_IP_LIST[-1]:
```

Comment:
May be better to do `i, ip in enumerate()`? In case the list somehow gets duplicate IPs.

Reply:
Hmmmm, I don't see why we would ever have duplicate IPs? But I can add.

Reply:
Done